src/test/base64.cc

```diff
@@ -98,3 +99,69 @@ TEST(FuzzEncoding, BadDecode2) {
   }
   ASSERT_EQ(failed, true);
 }
+
+void bench_base64_encode(size_t len, int encode_count) {
+  bufferlist bl;
+
+  int fd = open("/dev/urandom", O_RDONLY);
+  if (fd < 0) {
+    std::cout << "open /dev/urandom failed" << std::endl;
+    return;
+  }
+  bl.read_fd(fd, len);
+  EXPECT_EQ(bl.length(), len);
+
+  utime_t start = ceph_clock_now();
+  for (int i = 0; i < encode_count; i++) {
+    bufferlist rst;
+    bl.encode_base64(rst);
+  }
+  utime_t end = ceph_clock_now();
+  cout << "base64 encode len: " << setw(8) << len << ", "
+       << "encode " << setw(4) << encode_count << " times, "
+       << "total time(s): " << setw(8) << end - start << std::endl;
+}
+
+TEST(BenchBase64, BenchEncode) {
```

Comment:
i'd prefer not adding performance test in the middle of functional tests unless we can identify performance degradation in this way or it could help us to choose the right set of build-time or runtime parameters.